crates/miden-objects/src/asset/token_symbol.rs

```diff
@@ -1,20 +1,23 @@
-use alloc::string::String;
+use alloc::string::{String, ToString};
 
-use super::{AssetError, Felt};
+use super::{Felt, TokenSymbolError};
 
 #[derive(Default, Clone, Copy, Debug)]
 pub struct TokenSymbol(Felt);
 
 impl TokenSymbol {
     pub const MAX_SYMBOL_LENGTH: usize = 6;
-    pub const MAX_ENCODED_VALUE: u64 = 26u64.pow(TokenSymbol::MAX_SYMBOL_LENGTH as u32);
+    pub const ALPHABET_LENGTH: u64 = 26;
 
-    pub fn new(symbol: &str) -> Result<Self, AssetError> {
+    // this value is the result of encoding "ZZZZZZ" token symbol
+    pub const MAX_ENCODED_VALUE: u64 = 8031810156;
```

Comment:
Should we add a test to make sure we notice if that value goes out of sync somehow?

```rust
#[test]
fn test_token_symbol_max_value() {
    let token_symbol = TokenSymbol::try_from("ZZZZZZ").unwrap();
    assert_eq!(Felt::from(token_symbol).as_int(), TokenSymbol::MAX_ENCODED_VALUE);
}
```